audit_log/models.py

```diff
@@ -0,0 +1,11 @@
+from django.db import models
+from django.utils.translation import gettext as _
+
+
+class AuditLogEntry(models.Model):
+    is_sent = models.BooleanField(default=False, verbose_name=_("is sent"))
+    message = models.JSONField(verbose_name=_("message"))
+    created_at = models.DateTimeField(auto_now_add=True, verbose_name=_("created at"))
+
+    class Meta:
+        verbose_name_plural = "audit log entries"
```

Comment:
Here adding index to the message JSONField would making querying it according to its data fast, but it would increase the update/delete time (not important I think as audit logs are created more often than updated or deleted) and index size (I asked about this from Microsoft's copilot and it gave me an estimate of ~2–3 times the data size so if that is even remotely accurate number then it would mean that the GIN index would take not an insignificant space):

```suggestion
        verbose_name_plural = "audit log entries"
        indexes = [
            GinIndex(fields=["message"], name="message_gin_idx"),
        ]
```
And "python manage.py makemigrations" → `audit_log/management/migrations/0002_auditlogentry_message_gin_idx.py`:
```python
# Generated by Django 5.1.2 on 2024-11-15 11:36

import django.contrib.postgres.indexes
from django.db import migrations


class Migration(migrations.Migration):

    dependencies = [
        ('audit_log', '0001_initial'),
    ]

    operations = [
        migrations.AddIndex(
            model_name='auditlogentry',
            index=django.contrib.postgres.indexes.GinIndex(fields=['message'], name='message_gin_idx'),
        ),
    ]
```

So this is more like something we could discuss and see if there is a need for the indexing given the probably not insignificant index size it adds. Let's discuss!

Reply:
We talked about this and adding an index at this point is not perhaps necessary. If there come needs to query the audit log entries' message JSON data more efficiently, then an index can be added then. At the moment the audit log entries are more a dump temporary storage locally until they're pushed to ElasticSearch, not really meant to be queried that much.

Reply:
I added this as an example there